From 2133bca0bd1c060136f39435c64761756c0e7650 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Wed, 15 May 2019 12:27:35 +0530
Subject: [PATCH 271/386] octeontx2-pf: Remove vf flows cleanup functions

VF flows are deleted/disabled by AF during FLR or
NIX_LF_FREE message hence the existing functions which
send message to AF from PF to delete VF flows are removed.

Change-Id: I4ccff74af49cf9bc57afca85bdf04129415ddc16
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10405
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../marvell/octeontx2/nic/otx2_common.h       |  2 -
 .../marvell/octeontx2/nic/otx2_ethtool.c      | 53 -------------------
 .../ethernet/marvell/octeontx2/nic/otx2_pf.c  |  7 ---
 3 files changed, 62 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
index d7d6026fc3c3..5b5dc51bb983 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_common.h
@@ -517,8 +517,6 @@ void otx2vf_set_ethtool_ops(struct net_device *netdev);
 int otx2_install_rxvlan_offload_flow(struct otx2_nic *pfvf);
 int otx2_delete_rxvlan_offload_flow(struct otx2_nic *pfvf);
 int otx2_destroy_ethtool_flows(struct otx2_nic *pfvf);
-int otx2_delete_vf_ethtool_flows(struct otx2_nic *pfvf);
-int otx2_delete_ethtool_flows_for_vf(struct otx2_nic *pfvf, int vf);
 
 int otx2_open(struct net_device *netdev);
 int otx2_stop(struct net_device *netdev);
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index e7b3fdd01260..b3e1f75dc714 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -1448,56 +1448,3 @@ int otx2_destroy_ethtool_flows(struct otx2_nic *pfvf)
 
 	return 0;
 }
-
-int otx2_delete_vf_ethtool_flows(struct otx2_nic *pfvf)
-{
-	struct npc_delete_flow_req *req;
-	struct otx2_flow *iter, *tmp;
-	int err;
-
-	otx2_mbox_lock(&pfvf->mbox);
-	req = otx2_mbox_alloc_msg_npc_delete_flow(&pfvf->mbox);
-	if (!req) {
-		otx2_mbox_unlock(&pfvf->mbox);
-		return -ENOMEM;
-	}
-
-	req->all_vfs = 1;
-	/* Send message to AF */
-	err = otx2_sync_mbox_msg(&pfvf->mbox);
-	if (err) {
-		otx2_mbox_unlock(&pfvf->mbox);
-		return err;
-	}
-
-	otx2_mbox_unlock(&pfvf->mbox);
-	/* AF deleted VF entries now remove from ethtool list */
-	list_for_each_entry_safe(iter, tmp, &pfvf->flows, list) {
-		if (iter->is_vf) {
-			list_del(&iter->list);
-			kfree(iter);
-			pfvf->nr_flows--;
-		}
-	}
-
-	return 0;
-}
-
-int otx2_delete_ethtool_flows_for_vf(struct otx2_nic *pfvf, int vf)
-{
-	struct otx2_flow *iter, *tmp;
-	int err;
-
-	list_for_each_entry_safe(iter, tmp, &pfvf->flows, list) {
-		if (iter->vf != vf)
-			continue;
-		err = otx2_remove_flow_msg(pfvf, iter->entry, false);
-		if (err)
-			return err;
-		list_del(&iter->list);
-		kfree(iter);
-		pfvf->nr_flows--;
-	}
-
-	return 0;
-}
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 061855003a6e..bea628f2bbee 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -95,10 +95,6 @@ static void otx2_flr_handler(struct work_struct *work)
 
 	vf = flrwork - pf->flr_wrk;
 
-	/* Delete ethtool ntuple filters for the VF before FLR */
-	if (otx2_delete_ethtool_flows_for_vf(pf, vf + 1))
-		dev_err(pf->dev, "Unable to delete VF%d ntuple filters\n",
-			vf);
 
 	otx2_mbox_lock(&pf->mbox);
 	req = otx2_mbox_alloc_msg_vf_flr(&pf->mbox);
@@ -1712,8 +1708,6 @@ int otx2_stop(struct net_device *netdev)
 	/* First stop packet Rx/Tx */
 	otx2_rxtx_enable(pf, false);
 
-	otx2_destroy_ethtool_flows(pf);
-
 	pf->intf_down = true;
 	/* 'intf_down' may be checked on any cpu */
 	smp_wmb();
@@ -2412,7 +2406,6 @@ static int otx2_sriov_disable(struct pci_dev *pdev)
 	if (!pci_num_vf(pdev))
 		return 0;
 
-	otx2_delete_vf_ethtool_flows(pf);
 	pci_disable_sriov(pdev);
 
 	for (i = 0; i < pci_num_vf(pdev); i++)
-- 
2.17.1

